mono/metadata/object.c

```diff
@@ -3955,7 +3955,8 @@ mono_runtime_get_main_args (void)
 {
 	MONO_REQ_GC_UNSAFE_MODE;
 	MonoError error;
-	MonoArray *result = mono_runtime_get_main_args_checked (&error);
+	MonoArrayHandle *arg_array = mono_runtime_get_main_args_checked_handle (&error);
+	MonoArray *result = MONO_HANDLE_RAW (*arg_array);
 	mono_error_assert_ok (&error);
 	return result;
```

Comment:
So we don't want to leak the handle.   this function is a perfect candidate to be wrapped in `HANDLE_FUNCTION_ENTER()` / `HANDLE_FUNCTION_RETURN_OBJ (arg_array)`.